### PR TITLE
setup architecture

### PR DESCRIPTION
**Pull request by batchatT**

### CODE REVIEW

File: app/src/main/java/com/example/trempelapp/dataLayer/inMemory/UserInfoRepositoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.example.trempelapp.dataLayer.inMemory
+
+import com.example.trempelapp.dataLayer.inMemory.UserInfoRepository
+import javax.inject.Inject
+
+class UserInfoRepositoryImpl @Inject constructor(
+/*private val preferences: SharedPreferences*/
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/dataLayer/inMemory/UserInfoRepositoryImpl.kt:7]: Pass SharedPreferencesManager, And API service which was generated by retrofit

File: app/src/main/java/com/example/trempelapp/presentationLayer/splashScreen/TrempelSplashViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.example.trempelapp.presentationLayer.splashScreen
+
+import androidx.lifecycle.ViewModel
+import com.example.trempelapp.domainLayer.GetUserInfoUseCase
+import javax.inject.Inject
+
+class TrempelSplashViewModel @Inject constructor(
+    private val userInfoUseCase: GetUserInfoUseCase
+    ): ViewModel(){
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/presentationLayer/splashScreen/TrempelSplashViewModel.kt:9]: Create BaseViewModel and extend it

File: app/src/main/java/com/example/trempelapp/presentationLayer/logInScreen/LogInActivity.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.example.trempelapp.presentationLayer.logInScreen
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import com.example.trempelapp.R
+
+class LogInActivity : AppCompatActivity() {
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/presentationLayer/logInScreen/LogInActivity.kt:7]: Create both BaseActivity and BaseFragment and extend it

File: app/src/main/java/com/example/trempelapp/dataLayer/network/NetworkSource.kt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package com.example.trempelapp.dataLayer.network
+
+interface NetworkSource {
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/dataLayer/network/NetworkSource.kt:3]: I think it's not correct naming. You have to call it exactly what it should do. F.e. if this service is responsible for user endpoint call it either UserService or UserApi.

Comment (denisque75) [reply app/src/main/java/com/example/trempelapp/dataLayer/network/NetworkSource.kt:3]: And I cannot find Retrofit generator class. Did you miss it?

File: app/src/main/java/com/example/trempelapp/presentationLayer/splashScreen/TrempelSplashViewModel.kt (hunk 2 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/presentationLayer/splashScreen/TrempelSplashViewModel.kt:9]: done

File: app/src/main/java/com/example/trempelapp/dataLayer/inMemory/UserInfoRepositoryImpl.kt (hunk 1 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/dataLayer/inMemory/UserInfoRepositoryImpl.kt:7]: done

File: app/src/main/java/com/example/trempelapp/dataLayer/network/NetworkSource.kt (hunk 4 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/dataLayer/network/NetworkSource.kt:3]: done

File: app/src/main/java/com/example/trempelapp/presentationLayer/logInScreen/LogInActivity.kt (hunk 3 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/presentationLayer/logInScreen/LogInActivity.kt:7]: done

**Review by denisque75 — APPROVED** [review]

**Pull request merged**